src/NeoSharp.Core/Network/NetworkConfig.cs

```diff
@@ -1,16 +1,44 @@
-﻿using Microsoft.Extensions.Configuration;
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Configuration;
 
 namespace NeoSharp.Core.Network
 {
     public class NetworkConfig 
     {
-        public string ServerIp { get; set; }
+        private static readonly Regex _peerEndPointPattern = new Regex(@"^(?<proto>\w+)://(?<host>[^/:]+):?(?<port>\d+)?/?$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
 
-        public int ServerStartPort { get; set; }
+        public EndPoint[] PeerEndPoints { get; private set; }
 
         public NetworkConfig(IConfiguration configuration = null)
         {
-            configuration?.GetSection(nameof(NetworkConfig))?.Bind(this);
+            var peerEndPoints = configuration?.GetSection(nameof(NetworkConfig))?.GetSection("PeerEndPoints").Get<string[]>();
```

Comment:
Add null check at the end if is necessary please
ˋ?.Get<string[]>();ˋ

Reply:
Good point. All this ? in one line smell a bit.